src/bio/bio_config.c

```diff
@@ -773,66 +773,47 @@ decode_daos_data(const char *nvme_conf, const char *method_name, struct config_e
 	/* Get 'config' array first configuration entry */
 	ctx->config_it = spdk_json_array_first(ctx->config);
 	if (ctx->config_it == NULL) {
-		/* Entry not-found so return positive RC */
-		D_GOTO(out, rc = 1);
+		D_DEBUG(DB_MGMT, "%s entry not found in spdk bootstrap config\n", method_name);
+		D_GOTO(out, rc = JSON_NOT_FOUND);
 	}
 
 	while (ctx->config_it != NULL) {
 		rc = spdk_json_decode_object(ctx->config_it, config_entry_decoders,
-					     SPDK_COUNTOF(config_entry_decoders), cfg);
+					     SPDK_COUNTOF(config_entry_decoders), &cfg);
 		if (rc < 0) {
 			D_ERROR("Failed to decode 'config' entry: %s\n", spdk_strerror(-rc));
 			D_GOTO(out, rc = -DER_INVAL);
 		}
 
-		if (strcmp(cfg->method, method_name) == 0)
+		if (strcmp(cfg.method, method_name) == 0)
 			break;
 
 		/* Move on to next subsystem config */
 		ctx->config_it = spdk_json_next(ctx->config_it);
 	}
 
 	if (ctx->config_it == NULL) {
-		/* Entry not-found so return positive RC */
-		rc = 1;
+		D_DEBUG(DB_MGMT, "%s entry not found in spdk bootstrap config\n", method_name);
+		D_GOTO(out, rc = JSON_NOT_FOUND);
```

Comment:
Same remarks as line 777

Reply:
keeping as is so we have the same change on master and 2.6 branches